### DISCUSSION TITLE
Prespecified scenarios

### DISCUSSION THREAD

**Discussion by steipatr** [quaquel/EMAworkbench#321] [Q&A] [answered]
In my current line of work, I often find myself trying to run an EMA model with a small set of specific input parameter value combinations, rather than sampling across all input parameters independently. In the framework introduced in [Kwakkel & Haasnoot 2019](https://link.springer.com/chapter/10.1007/978-3-030-05252-2_15), these two options might be called prespecified scenarios and exploration, respectively. Right now I have to do this somewhat manually by using integer and/or categorical parameters to generate multiple data sets which are then merged. Could there be some more elegant way to do this, e.g. by defining the scenarios with dictionaries of input parameter names and values? E.g.

```
base_case = {
  "x1": 1,
  "x2": 10,
  "x3": 5
}

other_scenario = {
  "x1": 1.5,
  "x2": 10,
  "x3": 8
}
```

And then EMA runs the model only for those specific scenarios:

`results = perform_experiments(Model, scenarios = [base_case, other_scenario])`

 Is this too antithetical to the Workbench's underlying concepts?

**Comment by EwoutH** [comment]
I think we should be able to support that. It’s indeed not the core of the workbench, but I can think of some scenarios (pun intended) where you have pre-computed scenarios for whatever reason.

It might be even beneficial to split scenario generation and experiment execution in the workbench anyway. That might increase flexibility and help with reproducibility.

@quaquel what do you think?

**Comment by quaquel** [comment, marked as answer]
This is already supported. You can use the Scenario class for this directly. Roughly like this

```python
from ema_workbench import Scenario

scenario_1 = Scenario("base case", **parameter_dict)
scenario_2 = Scenario("other scenario", **some_other_dict)

perform_experiments(model, [scenario_1, scenario_2])


```

**Reply by steipatr** [reply]
I had no idea, thanks! Discussion can be closed from my side.
